operator/k8s/resource_ctors.go

```diff
@@ -57,3 +58,13 @@ func CiliumEndpointSliceResource(lc hive.Lifecycle, cs client.Clientset, opts ..
 	)
 	return resource.New[*cilium_api_v2alpha1.CiliumEndpointSlice](lc, lw, resource.WithMetric("CiliumEndpointSlice")), nil
 }
+
+func IngressClassResource(lc hive.Lifecycle, cs client.Clientset, opts ...func(*metav1.ListOptions)) (resource.Resource[*slim_networkingv1.IngressClass], error) {
+	if !cs.IsEnabled() {
```

Comment:
do we need an additional check that this resource is only initialized if ingress controller functionality is enabled?

Reply:
We could, but I don't know how helpful it would be? If ingress controller functionality isn't enabled, then the resource won't be marked as needed. I think that will prevent any goroutines from starting in the case that ingress functionality is disabled.

Reply:
leaving it as is is fine for me :+1: just wanted to mention it